### PR TITLE
Installer client-setup script: download pgo client binary

### PR DESCRIPTION
**Pull request by jmckulk**
This update will download the `pgo` (Linux) or `pgo-mac` (Darwin) binary from
github based on the operating system. The binary will be downloaded to
`~/.pgo/<operator-namespace>` and should be added to the users `PATH`.

**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

**What is the new behavior (if this is a feature change)?**

**Other information**:

### CODE REVIEW

File: installers/kubectl/client-setup.sh [hunk 1]
```diff
@@ -14,6 +14,40 @@
 # This script should be run after the operator has been deployed
 export PGO_OPERATOR_NAMESPACE="${PGO_OPERATOR_NAMESPACE:-pgo}"
 PGO_USER_ADMIN="${PGO_USER_ADMIN:-pgouser-admin}"
+PGO_CLIENT_VERSION="${PGO_CLIENT_VERSION:-v4.3.0}"
+PGO_CLIENT_URL="https://github.com/CrunchyData/postgres-operator/releases/download/${PGO_CLIENT_VERSION}"
+LINUX_BIN="pgo"
+DARWIN_BIN="pgo-mac"
+VALID_OS_OPTIONS="Linux, Darwin"
```
Comment (jkatz) [inline installers/kubectl/client-setup.sh:21]: This can just be hard coded. no need to have a variable

File: installers/kubectl/client-setup.sh [hunk 2]
```diff
@@ -14,6 +14,40 @@
...
+OUTPUT_DIR="${HOME}/.pgo/${PGO_OPERATOR_NAMESPACE}"
+mkdir -p "${OUTPUT_DIR}"
+# lock down the directory to just this user
+chmod a-rwx,u+rwx "${OUTPUT_DIR}"
+
+echo "Operating System found is ${UNAME_RESULT}. Downloading ${BIN_NAME} client binary..."
+
+FULL_PATH="${PGO_CLIENT_URL}/${BIN_NAME}"
+curl -L "${FULL_PATH}" -o "${OUTPUT_DIR}/${BIN_NAME}"
+chmod 0755 "${OUTPUT_DIR}/${BIN_NAME}"
```
Comment (jkatz) [inline installers/kubectl/client-setup.sh:47]: `chmod +x` will do it

File: installers/kubectl/client-setup.sh [hunk 3]
```diff
@@ -14,6 +14,40 @@
...
+
+echo "Binary downloaded to ${OUTPUT_DIR}/${BIN_NAME}, please update your bashrc with the following:"
+echo "export PATH=${OUTPUT_DIR}/${BIN_NAME}:\$PATH"
```
Comment (jkatz) [inline installers/kubectl/client-setup.sh:50]: I'd put this at the end with all the other stuff. Easy to copy and paste.

don't need the echo then to say it's downloaded.

File: installers/kubectl/client-setup.sh [hunk 4]
```diff
@@ -14,6 +14,40 @@
...
 PGO_USER_ADMIN="${PGO_USER_ADMIN:-pgouser-admin}"
+PGO_CLIENT_VERSION="${PGO_CLIENT_VERSION:-v4.3.0}"
+PGO_CLIENT_URL="https://github.com/CrunchyData/postgres-operator/releases/download/${PGO_CLIENT_VERSION}"
+LINUX_BIN="pgo"
+DARWIN_BIN="pgo-mac"
+VALID_OS_OPTIONS="Linux, Darwin"
+
+# Checks operating system and determines which binary to download
+UNAME_RESULT=$(uname)
+if [ "${UNAME_RESULT}" == "Linux" ]
```
Comment (cbandy) [inline installers/kubectl/client-setup.sh:25]: For single brackets (test) use `=`. For bash pattern matching, use `[[` and `==`.

Comment (jmckulk) [reply installers/kubectl/client-setup.sh:25]: Is this how it should be? I'm not sure i understand you comment
```
if [[ "${UNAME_RESULT}" == "Linux" ]]
then
    BIN_NAME="pgo"
elif [[ "${UNAME_RESULT}" == "Darwin" ]]
then
```

File: docs/content/installation/postgres-operator.md [hunk 5]
```diff
@@ -182,15 +182,19 @@ chmod +x client-setup.sh
 ```
 
 {{% notice tip %}}
-Running this script can cause existing `pgouser`, `client.crt`, and `client.key`
-files to be overwritten.
+Running this script can cause existing `pgo` binary, `pgouser`, `client.crt`,
```
Comment (jkatz) [inline docs/content/installation/postgres-operator.md:185]: `pgo` client executable,

**Pull request merged**